refactor/script

cdrani opened the pull request:
* refactor of the `js/script.js`
* Update copyright year
* reduce `5em` margin on `.card` to `2em`

cdrani committed:
reduce .card margin

cdrani committed:
update copyright year

cdrani committed:
refactor script.js

varlevi commented on css/styles.css:
```diff
@@ -117,7 +116,7 @@ input:focus {
   display: inline-block;
   padding: 30px;
-  margin: 5rem;
+  margin: 2rem;
```
The smaller margins look good in terms of width. I would, however, keep the larger margins on the top and bottom so that the cards are portrait oriented.

varlevi replied:
Sorry. Extra _padding_ needs to be added in order to make the card orientation portrait.
Maybe try something along the lines of `padding: 100px 60px;`?

varlevi committed:
Green Value Space Issue

The space between "Green" and "Value" in the placeholder for green input was actually a tab or new line. Changed back to space.

varlevi commented on index.html:
```diff
@@ -18,50 +18,49 @@
+      const darkmode = new Darkmode(options);
+      darkmode.showWidget();
+  </script>
 
```
Indentation fixes look good!

varlevi commented on index.html:
```diff
@@ -18,50 +18,49 @@
   <footer>
-  <p class="footer">Copyright &copy 2019 varlevi</p>
-</footer>
+    <p class="footer">Copyright &copy 2020 varlevi</p>
```
Maybe use JavaScript to automatically update this date every year?

cdrani replied:
Yup. Was thinking of that as well, but wanted to keep focus on the refactor only for this PR. Will update in a subsequent PR or just open a new issue with a `good first commit`/`beginners-only` label.

varlevi replied:
Great idea. I’ll open up that issue now.

varlevi commented on css/styles.css:
```diff
@@ -103,7 +103,6 @@ input:focus {
   height: 100px;
-  display: inline-block;
   border-radius: 15% 25%;
   margin: 15px;
```
Change this to:

`margin: 15px auto;`

in order to center the colors.

varlevi commented on js/script.js:
```diff
@@ -1,88 +1,100 @@
+        green += 2;
+    }
+
+    printContent(content);
```
These revisions definitely help in terms of code clarity and separation of data from the code. I think the organization and optimization of the code works really nicely as well. One thing that would be nice is clearer comments in the first section of code, however. Once I studied the code for a bit, it became pretty clear, but comment would certainly help to initially orient other developers to what's going on. Other than that, this look really nice! As always, thanks for your contribution.

cdrani replied:
Sure. I could add a blurb or two, but I feel like the function names are very descriptive as to their purpose.

RE: the huge margins on the cards, what do you thing of having the cards just be the width of animated colored box and without the `rgb()` text -- except maybe the base colour? In place of the text a tooltip would display it on hover. That way more cards can be shown be shown per row, they can be arranged using flexbox/grid for different size screens, and reduce the amount of vertical scrolling -- especially when there are a __lot__ of shades.

varlevi replied:
Hmm... I like the idea of removing the rgb code from above the color, but I’m thinking it might be better to have the user be able to click on the card and have the card expand to cover the screen and show the rgb code. The reason I say this is because while tooltips are great, they only work will really small chunks of text. At this point, the rgb code isn’t that big, but I have plans to add things like hex values and the like in the future, which would require more space. Creating something that is easy to add on to at this point seems like the best option for me, but I’d love to hear your thoughts on any other ideas. Also, I’m trying to keep the design look minimalistic, so while more shades per row is nice, I think some extra margin on either side of the page will be beneficial from a design viewpoint. Using flexbox or grid certainly sounds like a great way to do this. However, at this point I believe as separate issue and PR is needed, so I’ll open that up with some of the details we discussed if you approve.

varlevi reviewed (changes requested):
As I mentioned, clearer comments would be really helpful in the JS file and there were one or two things that needed fixed in the CSS file, but aside from that, everything looks greatly improved.

varlevi commented:
Alright. I've merged this into the beta branch now. I'll continue to review it and maybe add a thing or two, do some extra testing before I merge it into the master.

cdrani commented:
Sorry for not getting back to make these change requests. I usually only have time during the weekend to work on projects outside of work. 🙏

varlevi commented:
That’s fine. Totally understand. Thanks for all your contributions!
